test/integration/scheduler/workload_controller_test.go

```diff
@@ -69,6 +70,110 @@ var _ = ginkgo.Describe("Workload controller with scheduler", func() {
+				Obj()
+			gomega.Expect(k8sClient.Create(ctx, clusterQueue)).To(gomega.Succeed())
+
+			localQueue = testing.MakeLocalQueue("queue", ns.Name).ClusterQueue(clusterQueue.Name).Obj()
+			gomega.Expect(k8sClient.Create(ctx, localQueue)).To(gomega.Succeed())
+		})
+
+		ginkgo.AfterEach(func() {
+			gomega.Expect(util.DeleteNamespace(ctx, k8sClient, ns)).To(gomega.Succeed())
+			util.ExpectClusterQueueToBeDeleted(ctx, k8sClient, clusterQueue, true)
+			util.ExpectAdmissionCheckToBeDeleted(ctx, k8sClient, check3, true)
+			util.ExpectAdmissionCheckToBeDeleted(ctx, k8sClient, check2, true)
+			util.ExpectAdmissionCheckToBeDeleted(ctx, k8sClient, check1, true)
+			util.ExpectResourceFlavorToBeDeleted(ctx, k8sClient, flavor1, true)
+			util.ExpectResourceFlavorToBeDeleted(ctx, k8sClient, flavor2, true)
+		})
+
+		ginkgo.It("the workload should have appropriate AdditionalChecks added", func() {
+			wl := testing.MakeWorkload("wl", ns.Name).
+				Queue("queue").
+				Request(resourceGPU, "3").
+				Obj()
+			wlKey := client.ObjectKeyFromObject(wl)
+
+			ginkgo.By("creating and admitting the workload", func() {
+				gomega.Expect(k8sClient.Create(ctx, wl)).To(gomega.Succeed())
+				gomega.Eventually(func() bool {
+					if err := k8sClient.Get(ctx, client.ObjectKeyFromObject(wl), &updatedWorkload); err != nil {
+						return false
+					}
+					return workload.HasQuotaReservation(&updatedWorkload)
+				}, util.Timeout, util.Interval).Should(gomega.BeTrue())
+			})
+
+			ginkgo.By("checking workload's admission checks", func() {
+				gomega.Eventually(func() []string {
+					gomega.Expect(k8sClient.Get(ctx, wlKey, &updatedWorkload)).To(gomega.Succeed())
+					return slices.Map(updatedWorkload.Status.AdmissionChecks, func(c *kueue.AdmissionCheckState) string { return c.Name })
+				}, util.Timeout, util.Interval).Should(gomega.ConsistOf("check1", "check2"))
+			})
+
+			ginkgo.By("updating admission check strategies", func() {
+				createdQueue := kueue.ClusterQueue{}
+				queueKey := client.ObjectKeyFromObject(clusterQueue)
+				gomega.Eventually(func() error {
+					gomega.Expect(k8sClient.Get(ctx, queueKey, &createdQueue)).To(gomega.Succeed())
+					createdQueue.Spec.AdmissionCheckStrategy = []kueue.AdmissionCheckStrategy{
+						*testing.MakeAdmissionCheckStrategy("check1", flavorOnDemand).Obj(),
+						*testing.MakeAdmissionCheckStrategy("check2", reservationFlavor).Obj(),
+						*testing.MakeAdmissionCheckStrategy("check3").Obj()}
+					return k8sClient.Update(ctx, &createdQueue)
+				}, util.Timeout, util.Interval).Should(gomega.Succeed())
+			})
+
+			ginkgo.By("checking workload's admission checks", func() {
+				gomega.Eventually(func() []string {
+					gomega.Expect(k8sClient.Get(ctx, wlKey, &updatedWorkload)).To(gomega.Succeed())
+					return slices.Map(updatedWorkload.Status.AdmissionChecks, func(c *kueue.AdmissionCheckState) string { return c.Name })
+				}, util.Timeout, util.Interval).Should(gomega.ConsistOf("check1", "check3"))
+			})
+		})
```

Comment:
Complete the test by marking the checks as passed and verifying that at this point the Workload is admitted.

Reply:
Done

Reply:
Done